### PR TITLE
42 arduino i2c

### LINKED ISSUE #42: Arduino i2c library
Incorporate a library to help the main Arduino communicate with other Arduinos and Rpi.

### PR DESCRIPTION
**Pull request by nBukaty**
Started separating out responsibilities between the two Arduinos. Integrated our line follower code from last semester with i2c instead of writing to motors directly. 
Closes #42

### CODE REVIEW

File: arduino/drive/MotorArduino.ino [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  Serial.print(m_leftSpeed);
+  Serial.print(" ");
+  Serial.print(m_rightDirection);
+  Serial.print(" ");
+  Serial.print(m_rightSpeed);
+  Serial.println();
+  delay(100);
+}
+
+void receiveEvent(int bytesToRead){
```
Comment (Amertz08) [inline arduino/drive/MotorArduino.ino]: Space after parenthesis. Saw this multiple times. Only mentioning once.

File: arduino/sensors/linefollower/SensorArduino.ino [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  writeSpeed(motor_arduino_address, FWD, MAX_SPEED, FWD, 0);
+}
+
+uint8_t findLine(){
+  while(mySensorBar.getDensity() == 0){
+    writeSpeed(motor_arduino_address, BKWD, MAX_SPEED * 0.5, BKWD, MAX_SPEED * 0.5);
+  }
+  stopMotors();
+  if( mySensorBar.getPosition() < -50 )
+      {
```
Comment (Amertz08) [inline arduino/sensors/linefollower/SensorArduino.ino]: Indent is off

File: arduino/sensors/linefollower/SensorArduino.ino [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+void rightTurn(){
+  writeSpeed(motor_arduino_address, FWD, MAX_SPEED, FWD, 0);
+}
+
+uint8_t findLine(){
+  while(mySensorBar.getDensity() == 0){
+    writeSpeed(motor_arduino_address, BKWD, MAX_SPEED * 0.5, BKWD, MAX_SPEED * 0.5);
+  }
+  stopMotors();
+  if( mySensorBar.getPosition() < -50 )
```
Comment (Amertz08) [inline arduino/sensors/linefollower/SensorArduino.ino]: remove spacing

File: arduino/drive/MotorArduino.ino [hunk 4]
```diff
@@ -0,0 +1,39 @@
+#include <Wire.h>
```
Comment (Amertz08) [inline arduino/drive/MotorArduino.ino]: General comment. Think we have a consistent style guide to our code. This is my suggestion for C++

[Google C++](https://google.github.io/styleguide/cppguide.html)

**Comment by nBukaty** [comment]
Seeing as this specific code isn't done, just having added a feature, it makes the most sense to keep these things in mind and I'll correct them when adding sensor code this week.

**Comment by Amertz08** [comment]
Ok sounds good

**Comment by Amertz08** [comment]
Wait are you adding the sensor code to this PR or another one?

**Comment by Amertz08** [comment]
Ok thats fine. Could you make those minor formatting adjustments then I'll approve. Don't care to have a "Format Code" issue later down the line. Might as well be consistent early.

**Comment by nBukaty** [comment]
I'll just wait to merge it. I can fix it next time I sit down and work on
it.

**Comment by Amertz08** [comment]
Ok sounds good

**Pull request merged**